Sources/MongoSwift/AsyncAwait/MongoCollection+Indexes+AsyncAwait.swift

```diff
@@ -0,0 +1,207 @@
+    ) async throws {
+        try await self.dropIndex(keys, options: options, session: session).get()
+    }
+
+    /**
+     * Attempts to drop a single index from the collection given an `IndexModel` describing it.
+     *
+     * - Parameters:
+     *   - model: An `IndexModel` describing the index to drop.
+     *   - options: Optional `DropIndexOptions` to use for the command.
+     *   - session: Optional `ClientSession` to use when executing this command.
+     *
+     * - Throws:
+     *   - `MongoError.WriteError` if an error occurs while performing the command.
+     *   - `MongoError.CommandError` if an error occurs that prevents the command from executing.
+     *   - `MongoError.InvalidArgumentError` if the options passed in form an invalid combination.
+     *   - `MongoError.LogicError` if the provided session is inactive.
+     *   - `EncodingError` if an error occurs while encoding the options.
+     */
+    public func dropIndex(
+        _ model: IndexModel,
+        options: DropIndexOptions? = nil,
+        session: ClientSession? = nil
+    ) async throws {
+        try await self.dropIndex(model, options: options, session: session).get()
+    }
+
+    /**
+     * Drops all indexes in the collection.
+     *
+     * - Parameters:
+     *   - options: Optional `DropIndexOptions` to use for the command.
+     *   - session: Optional `ClientSession` to use when executing this command.
+     *
+     * - Throws:
+     *   - `MongoError.WriteError` if an error occurs while performing the command.
+     *   - `MongoError.CommandError` if an error occurs that prevents the command from executing.
+     *   - `MongoError.InvalidArgumentError` if the options passed in form an invalid combination.
+     *   - `MongoError.LogicError` if the provided session is inactive.
+     *   - `EncodingError` if an error occurs while encoding the options.
+     */
+    public func dropIndexes(
+        options: DropIndexOptions? = nil,
+        session: ClientSession? = nil
+    ) async throws {
+        try await self.dropIndexes(options: options, session: session).get()
+    }
+
+    /**
+     * Retrieves a list of the indexes currently on this collection.
+     *
+     * - Parameters:
+     *   - session: Optional `ClientSession` to use when executing this command.
+     *
+     * - Returns: A `MongoCursor` over the `IndexModel`s.
+     *
+     * - Throws: `MongoError.LogicError` if the provided session is inactive.
+     */
+    public func listIndexes(session: ClientSession? = nil) async throws -> MongoCursor<IndexModel> {
+        try await self.listIndexes(session: session).get()
+    }
+
+    /**
+     * Retrieves a list of names of the indexes currently on this collection.
+     *
+     * - Parameters:
+     *   - session: Optional `ClientSession` to use when executing this command.
+     *
+     * - Returns: A `MongoCursor` over the index names.
+     *
+     * - Throws: `MongoError.LogicError` if the provided session is inactive.
```

Comment:
General question, do we call out all possible errors that could be thrown in documentation? This method and the one above could encounter server/network errors e.g., but I'm not sure how that maps to our error types/if that's something we include in docs

Reply:
At some point, @patrickfreed went through the whole API and called out what we thought the most common errors would be. we generally haven't tried to list them exhaustively, I think the goal has generally been to focus on common pitfalls and errors that could be specific to the use of the particular API method.

there is an error handling guide in the docs that talks about some of the general common errors users could see on any method, like a `ConnectionError` or `ServerSelectionError`:
https://mongodb.github.io/mongo-swift-driver/docs/current/MongoSwift/error-handling-guide.html

Reply:
cool, makes sense to focus on errors that are relevant to usage. thanks for the clarification!